packages/generator-liferay-js/src/adapt/index.js

```diff
@@ -0,0 +1,260 @@
+		
+		Nevertheless, you can start with {'npm run deploy:liferay'} to deploy it
+		to your Liferay server.
+		`,
+	],
+	questions: `
+		It will be tuned accordingly, so that you can deploy it to your Liferay
+		server.
+
+		But first we need you to answer some customization questions...
+		`,
+	unsupportedProjectType: [
+		error`
+		Oops! Your project type is not supported by {Liferay JS Toolkit} or 
+		cannot be autodetected.
+		`,
+		info`
+		Please visit http://bit.ly/js-toolkit-wiki for the full list of 
+		supported project types and how they are detected.
+		`,
+	],
+	warnAboutOverwrite: [
+		'',
+		warn`
+		Now your project files will be modified. As a consequence, {Yeoman} may
+		notify you about the existence of a conflict and prompt for permission 
+		to overwrite your files.
+
+		Make sure to answer {'a'} or otherwise the adaptation to {Liferay JS 
+		Toolkit} will fail.
+		`,
+		info`
+		Note that you can also avoid this conflict warning providing the 
+		{'--force'} argument to Yeoman.
+		`,
+	],
+	welcome: [
+		title`
+		
+		|👋 |Welcome to Liferay JS Toolkit project adapter 
+		`,
+	],
+};
+
+/**
+ * Generator to add deploy support to projects.
+ */
+export default class extends Generator {
+	/**
+	 * Standard Yeoman initialization function
+	 */
+	initializing() {
+		this.sourceRoot(path.join(__dirname, 'templates'));
+
+		print(msg.welcome);
+
+		switch (project.probe.type) {
+			case project.probe.TYPE_CREATE_REACT_APP:
+				print(msg.createReactAppDetected);
+				this._tuneProject = () => this._tuneCreateReactAppProject();
+				break;
+
+			default:
+				print(msg.unsupportedProjectType);
+				process.exit(1);
+		}
+	}
+
+	/**
+	 * Standard Yeoman prompt function
+	 */
+	async prompting() {
+		print(msg.questions);
+
+		this.answers = {};
+
+		Object.assign(
+			this.answers,
+			await promptWithConfig(this, 'adapt', [
+				{
+					type: 'input',
+					name: 'category',
+					message:
+						'Under which category should your widget be listed?',
+					default: 'category.sample',
+				},
+			])
+		);
+
+		const answers = await promptWithConfig(this, 'adapt', [
+			{
+				type: 'confirm',
+				name: 'liferayPresent',
+				message:
+					'Do you have a local installation of Liferay for development?',
+				default: true,
+			},
+		]);
+
+		if (!answers.liferayPresent) {
+			return;
+		}
+
+		Object.assign(
+			this.answers,
+			await promptWithConfig(this, 'adapt', [
+				{
+					type: 'input',
+					name: 'liferayDir',
+					message:
+						'Where is your local installation of Liferay placed?',
+					default: '/liferay',
+					validate: validateLiferayDir,
+				},
+			])
+		);
+	}
+
+	/**
+	 * Standard Yeoman generation function
+	 */
+	writing() {
+		print(msg.warnAboutOverwrite);
+
+		this._copyTemplates();
+		this._modifyTemplates();
+		this._tuneProject();
+	}
+
+	/**
+	 * Standard Yeoman install function
+	 */
+	install() {
+		print(msg.projectAdapted);
+
+		this.installDependencies({
+			bower: false,
```

Comment:
If I'm not mistaken, `create-react-app` uses `yarn`,  so we should too.

```suggestion
			bower: false,
			npm: false,
			yarn: true,
```

Reply:
See https://yeoman.io/authoring/dependencies.html for reference